src/app/modules/main/chat_section/module.nim

```diff
@@ -836,8 +839,13 @@ proc updateTokenPermissionModel*(self: Module, permissions: Table[string, CheckP
 
   let adminRequirementMet = adminPermissions.len() > 0 and any(adminPermissions, proc (item: TokenPermissionItem): bool = item.tokenCriteriaMet)
 
-  let requiresPermissionToJoin = (adminPermissions.len() > 0 and adminRequirementMet) or memberPermissions.len() > 0
```

Comment:
I'm not sure if removing/changing this line is ok. What it did was show the permission to join screen even if there was just an admin permission.
Is it because we show the admin permissions as well?

Reply:
I just tested with an admin permission and it doesn't show, so I feel like the line is wrong. Or maybe it's a leftover from the time when we only asked for the shared address when there was a permission?

Reply:
yes, if you fit in one of the permissions - you must see the permission join screen, at least that is what I remember

Reply:
> yes, if you fit in one of the permissions - you must see the permission join screen, at least that is what I remember

Ok, I get the idea, but we currently don't show the other permissions, or maybe it's bugged? If we did show "if you hold X you will be an admin", I think it would make sense then.

Reply:
we show only permissions you fit in.
For example, admin permissions are hidden by default, but if you fit this permission - you'll see the requirements you are fit

Reply:
the same should be to TM but will it show something - I'm not sure :)

Reply:
Alright, since we do not show the TM token permission and we do the admin, I reverted for the admins, but I still made it so that if you have the TM token, you don't see the permission screen

I took that liberty because in the case of admin permissions, it makes sense to show the user that they own something special that makes them admin, but for TMs, it's useless to show them. That's because **only** the TM token makes you a TM, so if you own the TM token of a community, it means you know you'll be a TM in that community. So joining directly makes sense.

Without that change I made above, it created an un-intuitive screen where the button to join was blue and active, but the pill for the permission to join was red when you are TM. 

Obviously, the optimal solution would be to just show the TM token permission on that screen, but that requires more design, UI and backend work for a very un-needed feature as per my explanation above.